tests/data/meta_tensor/test_spatial_ndim.py

```
@@ -0,0 +1,139 @@
+# (shape, affine, expected_spatial_ndim)
+CONSTRUCTION_CASES = [
+    ((1, 10, 10, 10), None, 3),  # default eye(4)
+    ((1, 10, 10), torch.eye(3), 2),  # eye(3)
+    ((1, 10), torch.eye(2), 1),  # eye(2)
+]
+
+# (description, op, expected_spatial_ndim)  -- op takes a 2D MetaTensor and returns a new one
+PRESERVATION_CASES = [
+    ("reshape", lambda t: t.reshape(1, 100), 2),
+    ("unsqueeze", lambda t: t.unsqueeze(0), 2),
+    ("squeeze", lambda t: MetaTensor(torch.randn(1, 1, 10, 10), affine=torch.eye(3)).squeeze(1), 2),
```

Ruff (0.15.2)

[warning] Unused lambda argument: `t`

(ARG005)